client/src/app/components/Autocomplete.tsx

```diff
@@ -74,32 +87,82 @@ export const Autocomplete: React.FC<IAutocompleteProps> = ({
     if (!selections || selections.length === 0) {
       return [];
     }
-    return options.filter(
-      ({ id }) => selections.findIndex((s) => s.id === id) > -1
-    );
-  }, [options, selections]);
+    return isGrouped
+      ? options.filter((option) => {
+          return selections.some((selection) => {
+            return (
+              selection.id === option.id && selection.group === option.group
+            );
+          });
+        })
+      : options.filter((option) => {
+          return (
+            selections.findIndex((selection) => selection.id === option.id) > -1
+          );
+        });
+  }, [options, selections, isGrouped]);
 
   const filteredOptions = useMemo(() => {
-    return options.filter(
-      ({ id, name }) =>
-        selections.findIndex((s) => s.id === id) === -1 &&
-        toString(name).toLowerCase().includes(inputValue.toLocaleLowerCase())
-    );
-  }, [options, selections, inputValue]);
+    return options.filter((option) => {
+      const isOptionSelected = selections.some((selection) => {
+        const isSelectedById = selection.id === option.id;
+        const isSelectedByGroup = isGrouped
+          ? selection.group === option.group
+          : true;
+        return isSelectedById && isSelectedByGroup;
+      });
+
+      const isNameMatch = toString(option.name)
+        .toLowerCase()
+        .includes(inputValue.toLowerCase());
+      return !isOptionSelected && isNameMatch;
+    });
+  }, [options, selections, inputValue, isGrouped]);
+
+  const groupedOptions = useMemo((): GroupedOptions => {
+    if (!isGrouped) {
+      return {};
+    }
+
+    return filteredOptions.reduce((groups: GroupedOptions, option) => {
+      const groupName = option.group || "Ungrouped";
```

Comment:
`Ungrouped` is an unlikely group label but I would still avoid hardcoding.
Some ideas:
1. use empty string ?
2. drop the items without the group - they would need to be handled separately

Reply:
This part along with the ID field refactor piece feels like enough to push me toward just adding another grouped component.